util/conditions/getter.go

```diff
@@ -137,11 +137,47 @@ func Summary(from Getter, options ...MergeOption) *clusterv1.Condition {
 	return merge(conditionsInScope, clusterv1.ReadyCondition, mergeOpt)
 }
 
+// mirrorOptions allows to set options for the mirror operation.
+type mirrorOptions struct {
+	fallbackTo       *bool
+	fallbackReason   string
+	fallbackSeverity clusterv1.ConditionSeverity
+	fallbackMessage  string
+}
+
+// MirrorOptions defines an option for mirroring conditions.
+type MirrorOptions func(*mirrorOptions)
+
+// WithFallbackValue specify a fallback value to use in case the mirrored condition does not exists;
+// in case the fallbackValue is false, given values for reason, severity and message will be used.
+func WithFallbackValue(fallbackValue bool, reason string, severity clusterv1.ConditionSeverity, message string) MirrorOptions {
+	return func(c *mirrorOptions) {
+		c.fallbackTo = &fallbackValue
+		c.fallbackReason = reason
+		c.fallbackSeverity = severity
+		c.fallbackMessage = message
+	}
+}
+
 // Mirror mirrors the Ready condition from a dependent object into the target condition;
 // if the Ready condition does not exists in the source object, no target conditions is generated.
-func Mirror(from Getter, targetCondition clusterv1.ConditionType) *clusterv1.Condition {
+func Mirror(from Getter, targetCondition clusterv1.ConditionType, options ...MirrorOptions) *clusterv1.Condition {
```

Comment:
Should this be `MirrorReady`?

Reply:
changing this method to private and leaving only set methods

Reply:
done!